libraries/rust/data/src/client.rs

```diff
@@ -0,0 +1,218 @@
+use chrono::{DateTime, Utc};
+use reqwest::Url;
+use reqwest::header::{HeaderMap, HeaderValue, ACCEPT};
+use reqwest::Client as HTTPClient;
+use aws_sdk_s3::client::Client as S3Client;
+use aws_credential_types::Credentials;
+use aws_types::region::Region;
+use aws_sdk_s3::Config;
+use serde::{Serialize, Deserialize};
+use serde_json;
+use aws_sdk_s3::primitives::ByteStream;
+use flate2::write::GzEncoder;
+use flate2::read::GzDecoder;
+use flate2::Compression;
+use std::io::{Write, Read};
+
+#[derive(Deserialize)]
+struct BarsResponse {
+    bars: Vec<Bar>,
+    #[serde(rename = "next_page_token")]
+    next_token: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Bar {
+    pub ticker: Option<String>,
+    #[serde(rename = "t")]
+    pub timestamp: DateTime<Utc>,
+    #[serde(rename = "o")]
+    pub open: f32,
+    #[serde(rename = "h")]
+    pub high: f32,
+    #[serde(rename = "l")]
+    pub low: f32,
+    #[serde(rename = "c")]
+    pub close: f32,
+    #[serde(rename = "v")]
+    pub volume: u32,
+    #[serde(rename = "n")]
+    pub number_of_trades: u32,
+    #[serde(rename = "vw")]
+    pub volume_weighted_average_price: f32,
+}
+
+pub struct Client {
+    alpaca_base_url: String,
+    alpaca_api_key_id: String,
+    alpaca_api_secret_key: String,
+    http_client: HTTPClient,
+    s3_client: S3Client,
+    s3_data_bucket_name: String,
+}
```

Comment:
do you think it would make sense to have separate clients (we could combine them as well later) for cases where we may only want to connect to AWS, for example loading and transforming a s3 object and storing it back in s3?

Reply:
Possibly I just consolidated them into the same library since it'll at least initially only be used in a few data layer services.